source/client/process_impl.cc

```diff
@@ -260,11 +262,31 @@ void ProcessImpl::createBootstrapConfiguration(envoy::config::bootstrap::v2::Boo
         options_.maxPendingRequests() == 0 ? 1 : options_.maxPendingRequests());
     thresholds->mutable_max_requests()->set_value(options_.maxActiveRequests());
 
-    cluster->set_type(envoy::api::v2::Cluster::DiscoveryType::Cluster_DiscoveryType_STATIC);
-    auto* host = cluster->add_hosts();
-    auto* socket_address = host->mutable_socket_address();
-    socket_address->set_address(uri.address()->ip()->addressAsString());
-    socket_address->set_port_value(uri.port());
+    if (options_.backendEndpoints().empty()) {
+      cluster->set_type(envoy::api::v2::Cluster::DiscoveryType::Cluster_DiscoveryType_STATIC);
```

Comment:
nit: can you add a shorthand for this?  DiscoveryType isn't a namespace, but you could do either
```namespace EnvoyCluster = ::envoy::api::v2::Cluster;
EnvoyCluster::DiscoveryType::Cluster_DiscoveryType_STATIC```
or:
```const discovery_type = envoy::api::v2::Cluster::DiscoveryType;
discovery_type::Cluster_DiscoveryType_STATIC
```
I realize this part isn't new, but since you're now calling this long path at least 3 times, I think it'd help readability

Reply:
Removed this code in the updated version